scripts/lint-versions.js

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable no-console */
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+const { readdirSync, existsSync, readFileSync } = require('fs');
+
+const getDirectories = (source) =>
+    readdirSync(source, { withFileTypes: true })
+        .filter((pathMeta) => pathMeta.isDirectory())
+        .map((pathMeta) => pathMeta.name);
+
+function readPackageJsonDeps(filePath) {
+    if (existsSync(filePath)) {
+        const jsonData = JSON.parse(readFileSync(filePath, 'utf-8'));
+        const merged = {
+            ...jsonData.dependencies,
+            ...jsonData.devDependencies,
+        };
+        const result = {};
+        Object.keys(merged).forEach((dep) => {
+            if (merged[dep] && !merged[dep].includes('file:')) {
+                result[dep] = merged[dep];
+            }
+        });
+        return result;
+    }
+    return {};
+}
+
+function readPackageJsonNameVersion(filePath) {
+    if (existsSync(filePath)) {
+        const jsonData = JSON.parse(readFileSync(filePath, 'utf-8'));
+        const result = {};
+        result[jsonData.name] = `^${jsonData.version}`;
```

Comment:
sounds like this sets the expectation that the ^ matcher should be used consistently between packages, should we document this explicitly somewhere?

Reply:
Interesting. I think it's outlining we only confirm that versions match when the versions are not fixed or `~`. I think if you use either of those, you need to make a good reasoning for it in a PR, normally. Maybe that's over simplifying?

Happy to expand on the documentation here. What sort of messaging would make either of these realities clear?

Reply:
I think using ^ is a reasonable expectation, perhaps adding this script to a pre-commit hook would be sufficient to raise attention to this? Otherwise this seems like an easy thing to miss (in that unless a developer is aware of this script existing and knows what it does, they wouldn't know that ~ or fixed versions are not allowed).

Reply:
I added some documentation to that end. Do you think it would also be good to have it run on commit? I've not really run into this issue in this repo, so I'm not sure how to best apply it here.

Reply:
👍 i think what you have is fine as is, we can always add a hook later if it comes up frequently